### PR TITLE
Refs #12970 - sprockets-rails 2.3.3 for Patternfly

### PR DESCRIPTION
**Pull request by dLobatog**
https://github.com/theforeman/foreman-packaging/pull/944#issue-124499878

### CODE REVIEW

File: rubygem-sprockets-rails/rubygem-sprockets-rails.spec [hunk 1]
```diff
@@ -5,17 +5,17 @@
...
-Version: 2.2.4
-Release: 2%{?dist}
+Version: 2.3.3
+Release: 1%{?dist}
 Summary: Sprockets Rails integration
 Group: Development/Languages
 License: MIT
 URL: https://github.com/rails/sprockets-rails
-Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+Source0: http://rubygems.org/gems/%{gem_name}-%{version}.gem
```
Comment (ekohl) [inline rubygem-sprockets-rails/rubygem-sprockets-rails.spec]: Why doesn't https work?

Comment (dLobatog) [inline rubygem-sprockets-rails/rubygem-sprockets-rails.spec]: No idea, but when Jenkins runs wget it fails for rubygems https - http://ci.theforeman.org/job/packaging_build_rpm/3032/console

Comment (domcleal) [inline rubygem-sprockets-rails/rubygem-sprockets-rails.spec]: The Jenins slave that tito's usually run on is EL6 and it doesn't like those certificates, among other problems with HTTPS URLs.  It needs rebuilding.

**Comment by domcleal** [comment]
1) The -tests tarball needs to committed directly as a binary.  We can't use git-annex with a URL remote as there's nowhere to download it from unfortunately (and we're not using any storage with git-annex).

2) Change scl_prefix in the Requires/BuildRequires of the spec file to scl_prefix_ror.  I copied this spec file from the rh-ror41 collection to build only for Fedora so didn't fix it at the time.  Since we're building it in `tfm`, it needs to explicitly require the `ror` variants.

3) The commit message should say sprockets-rails, not sprockets.

4) foreman/foreman.spec will need to depend on `%{?scl_prefix}rubygem(sprockets-rails)` rather than the ror variant.

Hopefully tests will get further with those first two changes.

**Comment by dLobatog** [comment]
@domcleal Updated according to your comments, thanks!

**Comment by dLobatog** [comment]
Looks like tito attempts to `get` the tests tgz from `git-annex` and doesn't run them from the binary file. http://ci.theforeman.org/job/packaging_build_rpm/3050/console

File: rubygem-sprockets-rails/sprockets-rails-2.3.3-tests.tgz [hunk 2]
```diff
@@ -0,0 +1 @@
+../.git/annex/objects/fJ/qZ/SHA256E-s92435--64526351be8f06a44da4c019ae42a89e55a627373d9ffa5caa064bddf4fb6089.tgz/SHA256E-s92435--64526351be8f06a44da4c019ae42a89e55a627373d9ffa5caa064bddf4fb6089.tgz
```
Comment (domcleal) [inline rubygem-sprockets-rails/sprockets-rails-2.3.3-tests.tgz]: This is still an annexed file.

**Comment by dLobatog** [comment]
Duh, nevermind, I *did* git-annex the tests again.

**Comment by dLobatog** [comment]
Now tests are a binary again. I did `git reset HEAD~1` before and when I `git add`ed back the tests.tgz, it was added as an annex again.

File: foreman/foreman.spec [hunk 3]
```diff
@@ -190,6 +190,8 @@ BuildRequires: %{?scl_prefix_ror}rubygem(execjs) >= 1.4.0
 BuildRequires: %{?scl_prefix_ror}rubygem(execjs) < 2.5.0
 BuildRequires: %{?scl_prefix_ror}rubygem(jquery-rails) >= 3.1
 BuildRequires: %{?scl_prefix_ror}rubygem(jquery-rails) < 4.0
+BuildRequires: %{?scl_prefix_ror}rubygem(sprockets-rails) >= 2.3.3
```
Comment (domcleal) [inline foreman/foreman.spec]: a) scl_prefix as you're using the one in our collection
b) the older dependency on sprockets-rails needs removing, this replaces it

File: foreman/foreman.spec [hunk 4]
```diff
@@ -368,6 +370,8 @@ Requires: %{?scl_prefix_ror}rubygem(uglifier) >= 1.0.3
 Requires: %{?scl_prefix_ror}rubygem(execjs) >= 1.4.0
 Requires: %{?scl_prefix_ror}rubygem(jquery-rails) >= 3.1
 Requires: %{?scl_prefix_ror}rubygem(jquery-rails) < 4.0
+Requires: %{?scl_prefix_ror}rubygem(sprockets-rails) >= 2.3.3
```
Comment (domcleal) [inline foreman/foreman.spec]: ditto above.

Also http://projects.theforeman.org/issues/13074 should be fixed as having this dependency only in foreman-assets could lead to the wrong version being installed via the main foreman package.

File: rubygem-sprockets-rails/rubygem-sprockets-rails.spec [hunk 5]
```diff
@@ -5,31 +5,31 @@
...
+# git checkout v2.3.3
+# tar czvf sprockets-rails-2.3.3-tests.tgz test/
 Source2: sprockets-rails-%{version}-tests.tgz
 Requires: %{?scl_prefix_ruby}ruby(release)
 Requires: %{?scl_prefix_ruby}ruby(rubygems)
-Requires: %{?scl_prefix}rubygem(sprockets) => 2.8
-Requires: %{?scl_prefix}rubygem(sprockets) < 4
-Requires: %{?scl_prefix}rubygem(actionpack) >= 3.0
-Requires: %{?scl_prefix}rubygem(activesupport) >= 3.0
+Requires: %{?scl_prefix_ror}rubygem(sprockets) => 2.8
```
Comment (domcleal) [inline rubygem-sprockets-rails/rubygem-sprockets-rails.spec]: If you're packaging sprockets into our SCL too, then you need to change this to scl_prefix - ditto in BRs.

**Comment by dLobatog** [comment]
https://github.com/theforeman/foreman/pull/3047/files

**Comment by dLobatog** [comment]
@domcleal Rebased too

File: foreman/foreman.spec [hunk 6]
```diff
@@ -203,6 +203,8 @@ BuildRequires: %{?scl_prefix}rubygem(autoprefixer-rails) >= 5.2
 BuildRequires: %{?scl_prefix}rubygem(autoprefixer-rails) < 6.0
 BuildRequires: %{?scl_prefix}rubygem(patternfly-sass) >= 2.7
 BuildRequires: %{?scl_prefix}rubygem(patternfly-sass) < 3.0
+BuildRequires: %{?scl_prefix}rubygem(sprockets-rails) >= 2.3.3
```
Comment (domcleal) [inline foreman/foreman.spec]: This should be in the Gemfile section (it's already there, IIRC).

File: foreman/foreman.spec [hunk 7]
```diff
@@ -384,6 +386,8 @@ Requires: %{?scl_prefix}rubygem(autoprefixer-rails) >= 5.2
 Requires: %{?scl_prefix}rubygem(autoprefixer-rails) < 6.0
 Requires: %{?scl_prefix}rubygem(patternfly-sass) >= 2.7
 Requires: %{?scl_prefix}rubygem(patternfly-sass) < 3.0
+Requires: %{?scl_prefix}rubygem(sprockets-rails) >= 2.3.3
```
Comment (domcleal) [inline foreman/foreman.spec]: This should be in the main 'foreman' package's requires.

**Comment by dLobatog** [comment]
@domcleal Updated - now it's only on the regular Gemfile section & not on the assets package. It was like that because when I originally made this PR we hadn't moved sprockets & sprockets-rails to Gemfile yet.

File: foreman/foreman.spec [hunk 8]
```diff
@@ -186,8 +186,8 @@ BuildRequires: %{?scl_prefix}rubygem(protected_attributes) >= 1.1.1
 BuildRequires: %{?scl_prefix}rubygem(protected_attributes) < 1.2.0
 BuildRequires: %{?scl_prefix}rubygem(sprockets) >= 3
 BuildRequires: %{?scl_prefix}rubygem(sprockets) < 4
-BuildRequires: %{?scl_prefix_ror}rubygem(sprockets-rails) >= 2.2.2
-BuildRequires: %{?scl_prefix_ror}rubygem(sprockets-rails) < 3
+BuildRequires: %{?scl_prefix}rubygem(sprockets-rails) >= 2.2.2
```
Comment (domcleal) [inline foreman/foreman.spec]: 2.3.3?

**Comment by domcleal** [comment]
Also the requires on scl_prefix_ror for sprockets-rails in the main foreman spec file should be changed to scl_prefix, and updated to 2.3.3.

**Comment by dLobatog** [comment]
Updated, I realized they were ror and not tfm right after pushing :/

File: foreman/foreman.spec [hunk 9]
```diff
@@ -109,8 +109,8 @@ Requires: %{?scl_prefix}rubygem(protected_attributes) >= 1.1.1
 Requires: %{?scl_prefix}rubygem(protected_attributes) < 1.2.0
 Requires: %{?scl_prefix}rubygem(sprockets) >= 3
 Requires: %{?scl_prefix}rubygem(sprockets) < 4
-Requires: %{?scl_prefix_ror}rubygem(sprockets-rails) >= 2.2.2
-Requires: %{?scl_prefix_ror}rubygem(sprockets-rails) < 3
+Requires: %{?scl_prefix}rubygem(sprockets-rails) >= 2.2.3
```
Comment (domcleal) [inline foreman/foreman.spec]: `2.3.3` still :)

**Comment by dLobatog** [comment]
Duh, sorry for the typo, updated

**Comment by domcleal** [comment]
Thanks @dLobatog, building.

**Pull request merged**